src/Xamarin.Android.Build.Tasks/Tasks/Aapt.cs

```diff
@@ -331,9 +331,14 @@ protected string GenerateCommandLineCommands (string ManifestFile, string curren
 			if (AdditionalResourceDirectories != null)
 				foreach (var resdir in AdditionalResourceDirectories)
 					cmd.AppendSwitchIfNotNull ("-S ", resdir.ItemSpec.TrimEnd ('\\'));
-			if (AdditionalAndroidResourcePaths != null)
-				foreach (var dir in AdditionalAndroidResourcePaths)
-					cmd.AppendSwitchIfNotNull ("-S ", Path.Combine (dir.ItemSpec.TrimEnd (System.IO.Path.DirectorySeparatorChar), "res"));
+			if (AdditionalAndroidResourcePaths != null) {
+				foreach (var dir in AdditionalAndroidResourcePaths) {
+					var resdir = Path.Combine (dir.ItemSpec, "res");
+					if (Directory.Exists (resdir)) {
```

Comment:
@dellis1972 do you think it's worth adding a `Directory.Exists` check for every path passed via `-S`?

Reply:
I would say yes.